src/uct/cuda/cuda_copy/cuda_copy_md.h

```diff
@@ -18,33 +19,60 @@ typedef enum {
     UCT_CUDA_PREF_LOC_LAST
 } uct_cuda_pref_loc_t;
 
+typedef enum {
+    UCT_CUDA_ALLOC_METHOD_DEVICE,
+    UCT_CUDA_ALLOC_METHOD_FABRIC,
+    UCT_CUDA_ALLOC_METHOD_LAST
+} uct_cuda_alloc_method_t;
+
 
 /**
  * @brief cuda_copy MD descriptor
  */
 typedef struct uct_cuda_copy_md {
     struct uct_md               super;           /* Domain info */
+    unsigned                    sync_memops_set;
```

Comment:
if we init memops during md create, it is not needed

Reply:
@yosefe thought about this and I don't think we should init memops on md_open because cuda context creation may happen after ucp_init.

Reply:
indeed, so I think we should init context memops in UCM (do it once per each device[i])